shell/packages/sandstorm-db/profile.js

```diff
@@ -70,6 +70,26 @@ if (Meteor.isServer) {
 
   var Url = Npm.require("url");
   httpProtocol = Url.parse(process.env.ROOT_URL).protocol;
+
+  var Crypto = Npm.require("crypto");
+  Meteor.methods({
+    revealIdentity: function (token, identityId) {
+      var db = this.connection.sandstormDb;
+      if (!db.userHasIdentity(this.userId, identityId)) {
+        throw new Meteor.Error(403, "User doesn't own this indentity.");
+      }
+      var hashedToken = Crypto.createHash("sha256").update(token).digest("base64");
+      var ownerId = db.collections.apiTokens.findOne({_id: hashedToken}).accountId;
+      var profile = db.getIdentity(identityId).profile;
+      db.collections.contacts.upsert({ownerId: ownerId, identityId: identityId}, {
+        ownerId: ownerId,
+        petname: profile && profile.name,
+        created: new Date(),
+        identityId: identityId,
+        profile: profile,
```

Comment:
This will include `profile.pictureUrl`. You'll need to increment the refcount in `StaticAssets` if you want to avoid a potential 404 when the contact changes their picture.  (The refcounting is actually currently broken, but #1181 fixes that.)